fedmsg/__init__.py

```diff
@@ -62,9 +63,16 @@ def init(**kw):
 
 def API_function(doc=None):
     def api_function(func):
+        scrub = inspect.getargspec(func).args
 
         def _wrapper(*args, **kw):
             if not hasattr(__local, '__context'):
+                kw = kw.copy()
+
+                for arg in scrub:
+                    if arg in kw:
+                        del kw[arg]
```

Comment:
Heh, I don't think logger isn't set up at this point.  It would need its config which hasn't been loaded yet.